vm/src/obj/objbyteinner.rs

```diff
@@ -783,3 +1002,483 @@ pub enum ByteInnerPosition {
+    let mut splitted: Vec<&[u8]> = vec![];
+    let mut prev_index = slice.len();
+    let mut index = slice.len();
+    let mut count = 0;
+
+    // No sep given, will split for any \t \n \r and space  = [9, 10, 13, 32]
+    if sep.is_empty() {
+        //adjust index
+        index -= 1;
+
+        // rsplit without sep always trim right spaces for any maxsplit
+        // so we have to ignore right spaces.
+        loop {
+            if [9, 10, 13, 32].contains(&slice[index]) {
+                index -= 1
+            } else {
+                break;
+            }
+        }
+        prev_index = index + 1;
+
+        // most simple case
+        if maxsplit == 0 {
+            splitted.push(&slice[0..=index]);
+            return splitted;
+        }
+
+        // main loop. in_string means previous char is ascii char(true) or space(false)
+        // loop from right to left and reverse result the end
+        let mut in_string = true;
+        loop {
+            if [9, 10, 13, 32].contains(&slice[index]) {
+                if in_string {
+                    splitted.push(&slice[index + 1..prev_index]);
+                    count += 1;
+                    if count == maxsplit {
+                        // maxsplit reached, append, the remaing
+                        splitted.push(&slice[0..index]);
+                        break;
+                    }
+                    in_string = false;
+                    index -= 1;
+                    continue;
+                }
+            } else if !in_string {
+                in_string = true;
+                if index == 0 {
+                    splitted.push(&slice[0..1]);
+                    break;
+                }
+                prev_index = index + 1;
+            }
+            if index == 0 {
+                break;
+            }
+            index -= 1;
+        }
+    } else {
+        // sep is give, we match exact slice going backwards
+        while index != 0 {
+            if index <= sep.len() {
+                if &slice[0..index] == sep {
+                    splitted.push(&slice[index..prev_index]);
+                    splitted.push(&[]);
+                    break;
+                }
+                splitted.push(&slice[0..prev_index]);
+                break;
+            }
+            if &slice[(index - sep.len())..index] == sep {
+                splitted.push(&slice[index..prev_index]);
+                index -= sep.len();
+                prev_index = index;
+                count += 1;
+                if count == maxsplit {
+                    // maxsplit reached, append, the remaing
+                    splitted.push(&slice[0..prev_index]);
+                    break;
+                }
+                continue;
+            }
+
+            index -= 1;
+        }
+    }
+    splitted.reverse();
+    splitted
+}
+
+#[cfg(test)]
+
+// needed for dev. Same as python tests in bytes.py. should it be kept ?
```

Comment:
I would prefer the python test snippets for now. Using unit tests makes refactoring difficult. I think we are at the final iteration, but you never know if we want to change the internal design again, and then these tests will probably be deleted.